topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -473,6 +475,33 @@ public void setTopicCount(int topics) {
                 action.execute().onComplete(actionResult -> {
                     LOGGER.debug("{}: Executing handler for action {} on topic {}", logContext, action, lockName);
                     action.result = actionResult;
+
+                    String keytag = namespace + ":" + "KafkaTopic" + "/" + key.asKubeName().toString();
+                    Optional<Meter> metric = metrics.meterRegistry().getMeters()
+                            .stream()
+                            .filter(meter -> meter.getId().getName().equals(METRICS_PREFIX + "resource.state") &&
+                                    meter.getId().getTags().contains(Tag.of("kind", "KafkaTopic")) &&
+                                    meter.getId().getTags().contains(Tag.of("name", key.asKubeName().toString())) &&
+                                    meter.getId().getTags().contains(Tag.of("resource-namespace", namespace))
+                            ).findFirst();
+                    if (metric.isPresent()) {
+                        // remove metric so it can be re-added with new tags
+                        metrics.meterRegistry().remove(metric.get().getId());
+                        LOGGER.debug("Removed metric " + METRICS_PREFIX + "resource.state{}", keytag);
+                    }
```

Comment:
Is it necessary to remove it if it exists? I thought that `metrics.gauge` had the property that if the gauge already exists then the existing one is returned, and if not one is created. Which should mean there's no need for this find+remove logic at all.

Reply:
Iam not sure but I tried that and it was giving wrong behaviour. Like for eg. if a topic is working fine then it is with state 1 but suppose if we try to change the replicas in it the applied topic itself then the state should convert to since if we apply the chnages directly to applied topic then we get error. SO it was showing two states then, One was for previous one and one was for the new one

Reply:
Or we can also say it was showing the existing state too

Reply:
Oh, I guess this is because the tags are part of the gauge's identity, so in a case like that there would be two gauges: one with a reason and one without. That seems a bit weird to me. I feel that the gauge's identity should be 1-to-1 with a topic. But I can see that including the reason means you get a little more information when looking at the metric in grafana.

Reply:
@scholzj @sknot-rh are we happy with these semantics? Do other things in Strimzi do a similar thing?

Reply:
There is an issue https://github.com/strimzi/strimzi-kafka-operator/issues/4387. The removed resources keep appearing in this metric. Removing it solves this issue.

Reply:
So I think that means @ShubhamRwt should implement my suggestion, right @sknot-rh (that's what you meant)?